### DISCUSSION TITLE
Properties are reset after submit

### DISCUSSION THREAD

**Discussion by qwasyx0** [livewire/livewire#3918] [7. Bugs (archived)]
I'm having issues with submit data. After first submit the property gets reset to null. Here is the code:

The view:

```
<div class="px-6 py-4 bg-gray-200 overflow-hidden">
        <form wire:submit.prevent="submit">
                <input required name="variable" type="text" name="variable" id="variable" wire:model.defer="variable">
                <x-button>
                    {{ __('Submit') }}
                </x-button>
        </form>
 </div>
```

In the component:
```
  public $variable;

  public function submit()
    {
        dd($this->variable);
    }
```
Clicking on submit dye and dumps inputed value, but on next submit the value is null! This has now broken all my forms if there are some validation errors, in next submit all inputed fields disappear!

This is huge bug that I was debugging for atleast 6 hours, is anyone else having this issue?

I have since reverted to version 2.6.0 and everything works as it should, every next submit dye and dumps the same value.

**Comment by joshhanley** [comment]
@qwasyx0 can you please provide a small example repo that demonstrates the issue so we can investigate it?

Thanks!

**Reply by qwasyx0** [reply]
I have new finding, I think the issue is with dd(); without the function everything works as it should but with the function, clicking second time on submit the value is reseted to null. 

This is true even for 2.6.0 so i guess it's a feature that I found out just now.

**Reply by joshhanley** [reply]
@qwasyx0 yeah that's right, the reason is a `dd()` means dump and die, so Livewire doesn't run through all its other process which is required to ensure that the correct data is kept in sync.

Hope this helps!
